src/utils/molecule_utils.js

```diff
@@ -0,0 +1,89 @@
+import { Set as ISet } from 'immutable';
+
+const moleculeUtils = {
+  /**
+   * Given two arrays of ids, return true if they contain the same values in any order,
+   * ignoring duplicates
+   * @param idsA {Array}
+   * @param idsB {Array}
+   * @returns {Boolean}
+   */
+  compareIds(idsA, idsB) {
+    // If one array is empty and the other isn't, they can't contain the same values
+    if ((!idsA.length && idsB.length) || (idsA.length && !idsB.length)) {
```

Comment:
if this is the only way you're using immutable... you might as well just Object.freeze() and ref equality check, or do a set yourself (its in ES6), or just use arrays:

```
function eqSet(a, b) {
    if (a.length !== b.length) return false;
    //or a map if these are big...
    for (var c of a) if (b.indexOf(a) < 0) return false;
    return true;
}
```